beacon-chain/slasher/types/types.go

```diff
@@ -15,11 +15,19 @@ const (
 // for attester slashing detection.
 type CompactAttestation struct {
 	AttestingIndices []uint64
-	Source           uint64
-	Target           uint64
+	Source           types.Epoch
+	Target           types.Epoch
 	SigningRoot      [32]byte
 }
 
+// CompactBeaconBlock containing only the required information
+// for proposer slashing detection.
+type CompactBeaconBlock struct {
+	ProposerIndex uint64
+	Slot          uint64
+	SigningRoot   [32]byte
```

Comment:
Why do we need `SigningRoot`? 

If it's needed shouldn't we have multiple `SigningRoot`?

Reply:
this is a compact beacon block, not a proposer slashing. We need signing root to determine if a block is slashable with respect to another one, namely that the other fields match except for the signing root